File: DotNet/NUnitAssertionSamples/ClassWithExceptionsTest.cs
using System;
using CodeSamplesForAssertion;
using NUnit.Framework;

namespace NUnitAssertionSamples
{
    public class ClassWithExceptionsTest
    {
        [Test]
        public void ThrowsExceptionWhenNameIsNull()
        {
            var obj = new ClassWithExceptions();

            Assert.Throws<ArgumentNullException>(() => obj.SayHello(null));
        }

        [Test]
        public void ShouldHaveValidMessageForExceptionWhenNameIsNull()
        {
            var obj = new ClassWithExceptions();

            ArgumentNullException ex = Assert.Throws<ArgumentNullException>(() => obj.SayHello(null));

            Assert.AreEqual("Name cannot be null", ex.Message);
            //OR
            Assert.That(ex.Message, Is.EqualTo("Name cannot be null"));

            //OR above three lines could be replaced with:
            Assert.Throws(Is.TypeOf<ArgumentNullException>().And.Message.EqualTo("Name cannot be null"), () => obj.SayHello(null));


        }
    }
}

File: java/src/test/java/com/agile/CodeSamplesForAssertion/ClassWithExceptionsTest.java
package com.agile.CodeSamplesForAssertion;

import static org.junit.Assert.assertEquals;


import org.junit.Test;

public class ClassWithExceptionsTest {

    @Test(expected = NullPointerException.class)
    public void ThrowsExceptionWhenNameIsNull() throws Exception
        {
            ClassWithExceptions cwe = new ClassWithExceptions();
            String result =  cwe.sayHello(null);
            result.length();
           
        }
        

    @Test
    public void ShouldHaveValidMessageForExceptionWhenNameIsNull() throws Exception
    {
        String result="";
        try
    {
        ClassWithExceptions cwe = new ClassWithExceptions();
        result = cwe.sayHello("null");
    }
    catch( final NullPointerException e )
    {
        final String msg = "Name is Null";
        result.length();
        assertEquals(msg, e.getMessage());
    }
}


    @Test
    public void shouldReturnMessageWhenTheValidStringPassed() throws Exception
        {
            ClassWithExceptions cwe = new ClassWithExceptions();
             String result =  cwe.sayHello("abc");
            assertEquals("Hello abc",result);
           
        }
    


}